File: src/main/java/com/github/eendroroy/sdk/openweather/response/FindWeatherResponse.java
package com.github.eendroroy.sdk.openweather.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonRootName;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.List;

/**
 * @author indrajit
 */

@ToString
@Setter
@Getter
@JsonRootName("weathers")
public class FindWeatherResponse implements BaseResponse {
    @JsonProperty("cod")
    public Integer cod;

    @JsonProperty("message")
    public String message;

    @JsonProperty("count")
    public Integer count;


    @JsonProperty("list")
    public List<GetWeatherResponse> weathers;
}

File: src/main/kotlin/com/github/eendroroy/sdk/openweathermap/response/FindWeatherResponse.kt
package com.github.eendroroy.sdk.openweathermap.response

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonRootName

/**
 * @author indrajit
 */
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonRootName("weathers")
data class FindWeatherResponse(
        @JsonProperty("cod")     var cod: Int?,
        @JsonProperty("message") var message: String?,
        @JsonProperty("count")   var count: Int?,
        @JsonProperty("list")    var weathers: List<GetWeatherResponse>?
) : BaseResponse {
    override var httpCode: String? = null
    override var httpMessage: String? = null
}